File: src/main/java/licenta/service/UserService.java
package licenta.service;

import licenta.dto.UserRequestDTO;
import licenta.entity.User;
import licenta.entity.UserRole;

import java.util.List;
import java.util.Set;


public interface UserService {

    User createUser(UserRequestDTO user) throws Exception;

    User updateUser(UserRequestDTO user) throws Exception;

    User getUser(String username);

    List<User> getUsers();


    User getUserByUsername(String username);

    void deleteUser(Long userId);

    User updateUserRole(User user, Set<UserRole> userRoleSet);

    void addRecentlyViewedAnimal(Long userId, Long animalId);

    List<Long> getRecentlyViewedAnimals(Long userId);
}

File: frontend/licenta/src/app/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import baseUrl from "./helper";

@Injectable({
  providedIn: 'root'
})
export class UserService {
  constructor(private http: HttpClient) { }

  public addUser(user: any) {
    return this.http.post(`${baseUrl}/user/`, user);
  }

  public updateUserRole(username: string, role: string) {
    return this.http.put(`${baseUrl}/user/${username}/role/${role}`, null);
  }

  public getAllUsers() {
    return this.http.get(`${baseUrl}/user/`);
  }

  public updateUser(user: any) {
    return this.http.put(`${baseUrl}/user/`, user);
  }

  public deleteUser(userId: any) {
    console.log(userId)
    return this.http.delete(`${baseUrl}/user/${userId}`);
  }

  public addRecentlyViewedAnimal(userId: number, animalId: number) {
    return this.http.post(`${baseUrl}/user/${userId}/viewed/${animalId}`, null);
  }

  public getRecentlyViewedAnimals(userId: number) {
    return this.http.get(`${baseUrl}/user/${userId}/viewed`);
  }
}
